flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java

```diff
@@ -660,9 +720,30 @@ private TableSchema createTableSchema(SqlCreateTable sqlCreateTable) {
 			builder.watermark(rowtimeAttribute, getQuotedSqlString(validated), exprDataType);
 		});
 
+		// Set up table and column constraints into the schema.
+		for (SqlTableConstraint constraint : sqlCreateTable.getFullConstraints()) {
+			final String constraintName = constraint.getConstraintName();
+			if (constraint.isPrimaryKey()) {
+				if (constraintName != null) {
+					builder.primaryKey(constraintName, constraint.getColumnNames());
+				} else {
+					builder.primaryKey(constraint.getColumnNames());
+				}
+			}
+		}
 		return builder.build();
 	}
 
+	private void validateTableConstraint(SqlTableConstraint constraint) {
+		if (constraint.isUnique()) {
+			throw new ValidationException("UNIQUE constraint needs to implement");
+		}
+		if (constraint.isEnforced()) {
+			throw new ValidationException("ENFORCED mode needs to implement,"
```

Comment:
`Flink doesn't support ENFORCED mode for PRIMARY KEY constaint. ENFORCED/NOT ENFORCED controls if the constraint checks are performed on the incoming/outgoing data. Flink does not own the data therefore the only supported mode is NOT ENFORCED mode.`